plugin/pkg/webhook/webhook.go

```diff
@@ -70,28 +71,40 @@ func NewGenericWebhook(kubeConfigFile string, groupVersions []unversioned.GroupV
 	return &GenericWebhook{restClient, initialBackoff}, nil
 }
 
-// WithExponentialBackoff will retry webhookFn 5 times w/ exponentially
-// increasing backoff when a 429 or a 5xx response code is returned.
+// WithExponentialBackoff will retry webhookFn() up to 5 times with exponentially increasing backoff when
+// it returns an error for which apierrors.SuggestsClientDelay() or apierrors.IsInternalError() returns true.
 func (g *GenericWebhook) WithExponentialBackoff(webhookFn func() restclient.Result) restclient.Result {
+	var result restclient.Result
+	WithExponentialBackoff(g.initialBackoff, func() error {
+		result = webhookFn()
+		return result.Error()
+	})
+	return result
+}
+
+// WithExponentialBackoff will retry webhookFn() up to 5 times with exponentially increasing backoff when
+// it returns an error for which apierrors.SuggestsClientDelay() or apierrors.IsInternalError() returns true.
+func WithExponentialBackoff(initialBackoff time.Duration, webhookFn func() error) error {
 	backoff := wait.Backoff{
-		Duration: g.initialBackoff,
+		Duration: initialBackoff,
 		Factor:   1.5,
 		Jitter:   0.2,
 		Steps:    5,
 	}
-	var result restclient.Result
+
+	var err error
 	wait.ExponentialBackoff(backoff, func() (bool, error) {
-		result = webhookFn()
-		// Return from Request.Do() errors immediately.
-		if err := result.Error(); err != nil {
```

Comment:
`SuggestsClientDelay` returns true for StatusReasonServerTimeout and StatusReasonTimeout

NewGenericServerResponse translates StatusTooManyRequests (429) to StatusReasonTimeout - https://github.com/kubernetes/kubernetes/blob/0ca6506850d2055eb05fcb76e956d292c9f4ecf5/pkg/api/errors/errors.go#L322